### PR TITLE
Display meaningful error message when registering existing username

### LINKED ISSUE #5925: Form validation on login form exposes variables
if you try to signup with an existing username you get this

```
%(model_name)s with this %(field_label)s already exists.
```

### PR DESCRIPTION
**Pull request by unho**
Fixes #5925.

### CODE REVIEW

**Comment by dwaynebailey** [comment]
lgtm, but is there anything upstream that needs to be fixed?

**Comment by unho** [comment]
No, nothing needs fixing upstream.

This comes from our `User` model inheriting from `AbstractBaseUser` instead of `AbstractUser` (both Django abstracts, where `AbstractUser` inherits from `AbstractBaseUser`). `AbstractUser` includes this error message, but we don't inherit from that model to avoid having some unwanted fields in our `User` model.

**Comment by unho** [comment]
Reported failing interpolation in https://code.djangoproject.com/ticket/27898#ticket

File: pootle/apps/accounts/models.py [hunk 1]
```diff
@@ -60,6 +60,10 @@ class User(AbstractBaseUser):
                            _('Enter a valid username.'),
                            'invalid')
         ],
+        # Remove when fixed in Django: https://code.djangoproject.com/ticket/27898#ticket
```
Comment (dwaynebailey) [inline pootle/apps/accounts/models.py]: That's not true, we always want a human readable error here I'd assume.  The Django issue is about the interpolation which we will no longer use.

Comment (dwaynebailey) [inline pootle/apps/accounts/models.py]: Please try use something like `FIXME` as tools can flag that in future.  Apart from the fact that we don't need this comment at all.

**Comment by dwaynebailey** [comment]
Minor nits otherwise lgtm

File: pootle/apps/accounts/models.py (hunk 1 above)
Comment (unho) [inline pootle/apps/accounts/models.py]: k

**Pull request merged**